feat: remove external dependency

AndresXLP opened the pull request:
- Removed the Heimdall library; Raven Tree now natively manages timeouts and request retries without external dependencies.
- Add backoff strategy to manage retry requests.
- Add max backoff delay to prevent long delay backoff requests.
- Add err collections to handle multiple errors when retrying request.
- Remove server_test.go to test, now use httptest server.
- Remove log info
- Update README.md

AndresXLP committed:
feat: remove external dependency

- Removed the Heimdall library; Raven Tree now natively manages timeouts and request retries without external dependencies.
- Add backoff strategy to manage retry requests.
- Add max backoff delay to prevent long delay backoff requests.
- Add err collections to handle multiple errors when retrying request.
- Remove server_test.go to test, now use httptest server.
- Update README.md

jackgris commented on backoff.go:
```diff
@@ -0,0 +1,75 @@
+}
+
+// BackoffDefault initializes a BackoffStrategy with a default backoff delay.
+func BackoffDefault() *BackoffStrategy {
```
Quizás crear una función NewBackoff, con el patron "optional functional parameter"? Y reemplazar esas 3 funciones?

AndresXLP replied:
Muchas gracias Gabo seguro que lo aplicaré

jackgris commented on client_test.go:
```diff
@@ -44,25 +50,74 @@ type ravenTreeTestSuite struct {
+		suite.Equal(http.MethodGet, r.Method)
+
+		w.WriteHeader(http.StatusOK)
+		return
```
Ese return es redundante, si usas algo como `golangci-lint run` te lo muestra, y tiene sentido porque es una función que no retorna nada y termina ahí.

AndresXLP replied:
Listo voy a introducir el `golangci-lint` para ver si tengo más novedades por corregir

AndresXLP committed:
feat: add golangci-lint

- Add golangci-lint
- Solve issues golangci-lint
- Add badge golangci-lint & Go Report Card

AndresXLP committed:
feat: backoff with optional parameter

- Add optional parameters wo handle Backoff.
- Add backoff_test.go.
- Update client_test.go
- Update README.md
- Add golangci-lint to Github Actions

AndresXLP committed:
test: resolve time test

- add tolerance to time comparison

jackgris reviewed (approved):
Creo que quedo mucho mejor el repo, sin dependencias innecesarias y código mas limpio.
